internal/pkg/cli/recorder/recorder.go

```diff
@@ -127,6 +139,89 @@ func (r *Recorder) processData(mntns uint32) error {
 	return fmt.Errorf("find mntns %d in bpf data map", mntns)
 }
 
+func (r *Recorder) generateAppArmorProfile() apparmorprofileapi.AppArmorAbstract {
+	processed := r.bpfRecorder.GetAppArmorProcessed()
+
+	abstract := apparmorprofileapi.AppArmorAbstract{}
+	enabled := true
+	var gotExec bool
+
+	exec := apparmorprofileapi.AppArmorExecutablesRules{}
+	if len(processed.FileProcessed.AllowedExecutables) != 0 {
+		sort.Strings(processed.FileProcessed.AllowedExecutables)
+		exec.AllowedExecutables = &processed.FileProcessed.AllowedExecutables
+		gotExec = true
+	}
+	if len(processed.FileProcessed.AllowedLibraries) != 0 {
+		sort.Strings(processed.FileProcessed.AllowedLibraries)
+		exec.AllowedLibraries = &processed.FileProcessed.AllowedLibraries
+		gotExec = true
+	}
+
+	if gotExec {
+		abstract.Executable = &exec
+	}
+
+	if (len(processed.FileProcessed.ReadOnlyPaths) != 0) ||
+		(len(processed.FileProcessed.WriteOnlyPaths) != 0) ||
+		(len(processed.FileProcessed.ReadWritePaths) != 0) {
+		files := apparmorprofileapi.AppArmorFsRules{}
+		if len(processed.FileProcessed.ReadOnlyPaths) != 0 {
+			sort.Strings(processed.FileProcessed.ReadOnlyPaths)
```

Comment:
nit: maybe we should do a deep copy into CR everywhere instead of doing shallow copy. Shallow copy sometimes can cause issues when the CR gets updated.